junto-rust/zomes/core/code/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+#![feature(try_from)]
+#[macro_use]
+extern crate hdk;
+extern crate serde;
+#[macro_use] 
+extern crate maplit;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+#[macro_use]
+extern crate holochain_core_types_derive;
+extern crate chrono;
+extern crate regex;
+
+use multihash::Hash;
+use hdk::{
+    api::DNA_ADDRESS,
+    error::ZomeApiResult,
+    holochain_core_types::{
+        json::JsonString, 
+        cas::content::Address,
+        hash::HashString,
+        error::HolochainError
+    }
+};
+
+mod expressions;
+
+use expressions::definitions::app_definitions;
+use expressions::definitions::function_definitions;
+
+define_zome! {
+    entries: [
+        expressions::definitions::user_entry_definitions::user_name_definition(),
+        expressions::definitions::user_entry_definitions::user_definition(),
+        expressions::definitions::time_entry_definitions::time_definiton(),
+        expressions::definitions::channel_entry_definitions::channel_definition(),
+        expressions::definitions::group_entry_definitions::group_definition(),
+        expressions::definitions::post_entry_definitions::post_definition()
+        //expressions::definitions::post_entry_definitions::resonation_definition()
+    ]
+
+    genesis: || { 
+        {
+            expressions::channel::create_collective_channel(&Address::from(hdk::api::DNA_ADDRESS.to_string()))?;
```

Comment:
Be careful trying to author entries in genesis. When the genesis callback is running the agent has no local chain. It is a pure function (like a validation function) that determines if a particular agent is able to join the hApp. Not a place to do setup. In future it will have more data available to it (e.g. the agent_address, any startup arguments the conductor passes like an access token) but authoring entries will never be allowed.

We are going to add another callback `initialize` which will happen after genesis if successful. The agent will then have a local chain and be able to author initial entries. If this is something you really need be sure to make some noise about it on the core repo so it happens :P